src/Frontend/Components/ImportDialog/ImportDialog.tsx

```
@@ -0,0 +1,258 @@
+      case FilePathValidity.VALIDATION_FAILED:
+        return 'Invalid file path';
+      default:
+        return null;
+    }
+  }, [opossumFilePathValidity]);
+
+  const opossumFilePathWarnMessage =
+    opossumFilePathValidity === FilePathValidity.OVERWRITE_WARNING
+      ? 'Warning: A file already exists at this location. If you continue, the existing file will be overwritten!'
+      : null;
+
+  function updateInputFilePath(filePath: string) {
+    setInputFilePath(filePath);
+    if (!opossumFilePathEdited) {
+      const derivedOpossumFilePath = getDotOpossumFilePath(
+        filePath,
+        fileFormat[1],
+      );
+      setOpossumFilePath(derivedOpossumFilePath);
+    }
+  }
+
+  function updateOpossumFilePath(filePath: string) {
+    setOpossumFilePath(filePath);
+  }
+
+  function editOpossumFilePath(filePath: string) {
+    updateOpossumFilePath(filePath);
+    if (filePath) {
+      setOpossumFilePathEdited(true);
+    } else {
+      setOpossumFilePathEdited(false);
+    }
+  }
+
+  function selectInputFilePath(): void {
+    window.electronAPI.importFileSelectInput(fileFormat).then(
+      (filePath) => {
+        if (filePath) {
+          updateInputFilePath(filePath);
+        }
+      },
+      () => {},
+    );
+  }
+
+  function selectOpossumFilePath(): void {
+    const defaultPath = opossumFilePath || 'imported.opossum';
+    window.electronAPI.importFileSelectSaveLocation(defaultPath).then(
+      (filePath) => {
+        if (filePath) {
+          editOpossumFilePath(filePath);
+        }
+      },
+      () => {},
+    );
+  }
+
+  function onCancel(): void {
+    dispatch(closePopup());
+  }
+
+  async function onConfirm(): Promise<void> {
+    setIsLoading(true);
+
+    setShowInputFilePathErrors(true);
+    setShowOpossumFilePathErrors(true);
+
+    if (inputFilePathIsValid && opossumFilePathIsValid) {
+      const success = await window.electronAPI.importFileConvertAndLoad(
```

can you return a possible error message here? then we don't need to make another BE call to validate the inputs. after all, validating the inputs could be unrelated to why the confirmation step was unsuccessful.

---

error messages now processed through error logs sent from backend